### PR TITLE
Made error reporting in unity sframe append more robust.

### PR DESCRIPTION
**Pull request by hoytak**

### CODE REVIEW

File: src/unity/lib/unity_sframe.cpp [hunk 1]
```diff
@@ -844,20 +844,50 @@ std::shared_ptr<unity_sframe_base> unity_sframe::append(
...
+                          other_column_names.begin(), other_column_names.end(),
+                          std::inserter(in_this, in_this.begin()));
+
+      std::set_difference(other_column_names.begin(), other_column_names.end(),
+                          column_names.begin(), column_names.end(),
+                          std::inserter(in_other, in_other.begin()));
+
+        std::ostringstream ss; 
+          ss << "Error: Attempting to append SFrame with columns "
+            << in_other << " to SFrame with columns " << in_this << ".";
```
Comment (znation) [inline src/unity/lib/unity_sframe.cpp]: Do we want to use the set difference for `in_this`? I would expect as a user to get the full list of columns of the SFrame being appended to (including the intersection with `in_other`).

Comment (znation) [reply src/unity/lib/unity_sframe.cpp]: Or maybe simplify it to the same message style as for types below (just show the names of columns in the `other_sframe` that aren't valid in `this` one)?

Comment (hoytak) [reply src/unity/lib/unity_sframe.cpp]: Fair point; this just reports the mismatch.   What about this?  "Columns [....] in base SFrame not found in appending SFrame."  ?

Comment (znation) [reply src/unity/lib/unity_sframe.cpp]: That sounds good.

**Comment by hoytak** [comment]
@znation -- could you review the new changes? 

Thanks!

**Pull request merged**